nipype/utils/nipype2boutiques.py

```diff
@@ -473,12 +473,10 @@ def get_boutiques_output(outputs, name, spec, interface, tool_inputs):
         output["list"] = True
         if output_value:
             # Check if all extensions are the same
-            extensions = []
-            for val in output_value:
-                extensions.append(os.path.splitext(val)[1])
+            extensions = {os.path.splitext(val)[1] for val in output_value}
             # If extensions all the same, set path template as
             # wildcard + extension. Otherwise just use a wildcard
-            if len(set(extensions)) == 1:
+            if len(extensions) == 1:
                 output["path-template"] = "*" + extensions[0]
```

Comment:
Now that `extensions` is a set, we can't index it:

```suggestion
                output["path-template"] = "*" + extensions.pop()
```